rs/sns/governance/src/governance.rs

```diff
@@ -4935,6 +4938,18 @@ impl Governance {
                 } else {
                     neuron.maturity_e8s_equivalent += neuron_reward_e8s;
                 }
+                // Insert the neuron_reward_e8s into the reward_events_to_neuron_reward_e8s map on the neuron. 
+                neuron.reward_events_to_neuron_reward_e8s.insert( 
+                    reward_event_end_timestamp_seconds, 
+                    neuron_reward_e8s
+                );
+                // prune
+                while neuron.reward_events_to_neuron_reward_e8s.len() > MAX_KEEP_REWARD_EVENTS_TO_NEURON_REWARD_E8S_PER_NEURON as usize {
```

Comment:
:+1: I wasn't sure if I could rely on the protobuf-to-rust conversion always generating a BTreeMap but now using map.pop_first is only possible on a BTreeMap so if the protobuf-to-rust conversion ever changes to a HashMap then it will be noticed and handled. 

It is cool how you choose to prune the map before insertion, good to get that perspective.